src/model/range.js

```diff
@@ -804,13 +829,13 @@ export default class Range {
 		// 4. At this moment we don't need the original range.
 		// We are going to modify the result and we need to return a new instance of Range.
 		// We have to create a copy of the reference range.
-		const result = new this( ref.start, ref.end );
+		let result = new this( ref.start, ref.end );
```

Comment:
AFAICS, range `result` is not used directly, just its `start` and `end` properties. Maybe we could just go with storing `Position`s and it would be better?